### DISCUSSION TITLE
How to sample mtsdf? and few other questions..

### DISCUSSION THREAD

**Discussion by BULLSEYEliTe** [Chlumsky/msdfgen#156] [Q&A]
Excuse my lack of information on this topic & It maybe oversight from my part but I couldn't find any resource on how to sample mtsdf's , the median method works well for msdf as shown but can't figure mtsdf.
2- Is it possible to incorporate this library into unreal engine and create sdfs from textures in editor or/and runtime?

3- Are supported input ttf, svg only? what about png file as input? or do we need to change png to ttf first?

Thank you for your help.

**Comment by Chlumsky** [comment]
MTSDF mode simply combines a MSDF and a plain SDF. Sample the RGB channels to use it as a MSDF (pseudo-distance metric) and sample the alpha channel if you need the true distance but without sharp corners (e.g. for some effects). If you don't know how to use it, you probably don't need it because it's for some more specialized use cases.

2. I don't use Unreal Engine so no idea if you can incorporate it but you can't create SDF from image textures (see 3).
3. No, PNG file cannot be used as input because MSDF is computed from vector geometry. You can compute (much less precise) plain SDF from an image but not with this tool.
